lib/ansible/modules/network/fortios/fortios_ipv4_policy.py

```diff
@@ -0,0 +1,324 @@
+            result['changed'] = True
+
+    #Present state
+    if module.params['state'] == 'present':
+        new_policy = FortiConfig(module.params['id'], 'edit')
+
+        #src / dest / service / interfaces
+        new_policy.set_param('srcintf', '"{0}"'.format(module.params['src_intf']))
+        new_policy.set_param('dstintf', '"{0}"'.format(module.params['dst_intf']))
+
+
+        new_policy.set_param('srcaddr', " ".join('"' + item + '"' for item in module.params['src_addr']))
+        new_policy.set_param('dstaddr', " ".join('"' + item + '"' for item in module.params['dst_addr']))
+        new_policy.set_param('service', " ".join('"' + item + '"' for item in module.params['service']))
+
+        # negate src / dest / service
+        if module.params['src_addr_negate']:
+            new_policy.set_param('srcaddr-negate', 'enable')
+        if module.params['dst_addr_negate']:
+            new_policy.set_param('dstaddr-negate', 'enable')
+        if module.params['service_negate']:
+            new_policy.set_param('service-negate', 'enable')
+
+        # action
+        new_policy.set_param('action', '{0}'.format(module.params['action']))
+
+        # Schedule
+        new_policy.set_param('schedule', '"{0}"'.format(module.params['schedule']))
+
+        #NAT
+        if module.params['nat']:
+            new_policy.set_param('nat', 'enable')
+            if module.params['fixedport']:
+                new_policy.set_param('fixedport', 'enable')
+            if module.params['poolname'] is not None:
+                new_policy.set_param('ippool', 'enable')
+                new_policy.set_param('poolname', '"{0}"'.format(module.params['poolname']))
+
+        #security profiles:
+        if module.params['av_profile'] is not None:
+            new_policy.set_param('av-profile', '"{0}"'.format(module.params['av_profile']))
+        if module.params['webfilter_profile'] is not None:
+            new_policy.set_param('webfilter-profile', '"{0}"'.format(module.params['webfilter_profile']))
+        if module.params['ips_sensor'] is not None:
+            new_policy.set_param('ips-sensor', '"{0}"'.format(module.params['ips_sensor']))
+        if module.params['application_list'] is not None:
+            new_policy.set_param('application-list', '"{0}"'.format(module.params['application_list']))
+
+        # comment
+        if module.params['comment'] is not None:
+            new_policy.set_param('comment', '"{0}"'.format(module.params['comment']))
+
+        #add to candidate config
+        f.candidate_config[path][module.params['id']] = new_policy
+
+        #check if change needed
+        change_string = f.compare_config()
+
+        if change_string != "":
+            result['change_string'] = change_string
+            result['changed'] = True
+
+    #Commit if not check mode
+    if module.check_mode is False and change_string != "":
+        try:
+            f.commit()
+        except FailedCommit:
+            #rollback
```

Comment:
pyFG auto start rollback in case of failure
https://github.com/spotify/pyfg/blob/master/pyFG/fortios.py line 330.
(allready tested a lot while coding ^^)